bin/luminork/src/args.rs

```diff
@@ -262,149 +253,168 @@ impl Args {
+    }
+    if let Some(module_index_url) = args.module_index_url {
+        config_map.set("module_index_url", module_index_url);
+    }
 
-            if let Some(layer_cache_disk_path) = args.layer_db_disk_path {
-                config_map.set("layer_db_config.disk_path", layer_cache_disk_path);
-            }
-            if let Some(pkgs_path) = args.pkgs_path {
-                config_map.set("pkgs_path", pkgs_path);
-            }
-            if let Some(module_index_url) = args.module_index_url {
-                config_map.set("module_index_url", module_index_url);
-            }
+    if let Some(auth_api_url) = args.auth_api_url {
+        config_map.set("auth_api_url", auth_api_url);
+    }
 
-            if let Some(auth_api_url) = args.auth_api_url {
-                config_map.set("auth_api_url", auth_api_url);
-            }
+    config_map.set("boot_feature_flags", args.features);
 
-            config_map.set("boot_feature_flags", args.features);
+    if let Some(create_workspace_permissions) = args.create_workspace_permissions {
+        config_map.set("create_workspace_permissions", create_workspace_permissions);
+    }
 
-            if let Some(create_workspace_permissions) = args.create_workspace_permissions {
-                config_map.set("create_workspace_permissions", create_workspace_permissions);
-            }
+    if !args.create_workspace_allowlist.is_empty() {
+        config_map.set(
+            "create_workspace_allowlist",
+            args.create_workspace_allowlist,
+        );
+    }
 
-            if !args.create_workspace_allowlist.is_empty() {
-                config_map.set(
-                    "create_workspace_allowlist",
-                    args.create_workspace_allowlist,
-                );
-            }
+    config_map.set("nats.connection_name", NAME);
+    config_map.set("pg.application_name", NAME);
+    config_map.set("layer_db_config.pg_pool_config.application_name", NAME);
+    config_map.set("layer_db_config.nats_config.connection_name", NAME);
+    config_map
+}
+
+impl TryFrom<Args> for Config {
+    type Error = ConfigError;
 
-            config_map.set("nats.connection_name", NAME);
-            config_map.set("pg.application_name", NAME);
-            config_map.set("layer_db_config.pg_pool_config.application_name", NAME);
-            config_map.set("layer_db_config.nats_config.connection_name", NAME);
+    fn try_from(args: Args) -> Result<Self, Self::Error> {
+        ConfigFile::layered_load(NAME, |config_map| {
+            build_config_map(args, config_map);
         })?
         .try_into()
     }
 }
 
+pub async fn load_config_with_provider<P>(
+    args: Args,
+    provider: Option<(P, String)>,
+) -> Result<Config, ConfigError>
+where
+    P: ParameterProvider + 'static,
+{
+    ConfigFile::layered_load_with_provider::<_, P>(NAME, provider, move |config_map| {
+        build_config_map(args, config_map);
+    })
+    .await?
+    .try_into()
+}
+
```

Comment:
This divergent behavior won't be repeated with the other clients once we verify this pattern works with luminork. We'll just replace the `try` calls entirely.